src/historywork/WriteVerifiedCheckpointHashesWork.cpp

```diff
@@ -81,6 +131,14 @@ WriteVerifiedCheckpointHashesWork::~WriteVerifiedCheckpointHashesWork()
 bool
 WriteVerifiedCheckpointHashesWork::hasNext() const
 {
+    if (mFromLedger)
+    {
+        return mCurrCheckpoint >= *mFromLedger;
+    }
+    else if (mLatestTrustedHashPair)
+    {
+        return mCurrCheckpoint > mLatestTrustedHashPair->first;
```

Comment:
Should this be `return mCurrCheckpoint >= mLatestTrustedHashPair->first;`? Currently I don't think we actually check the hash consistency from `mLatestTrustedHashPair.ledgerSeq` to `mLatestTrustedHashPair.ledgerSeq + 1`.

Reply:
Correct, when validating down to mLatestTrustedHashPair, this won't check the hash consistency between `mLatestTrustedHashPair.ledgerSeq` and `mLatestTrustedHashPair.ledgerSeq + 1`  if mLatestTrustedHashPair.ledgerSeq is a checkpoint boundary (which it ought to be). 

I think this is fine, as the user trusts the input already and the output will contain all the subsequent hashes based in the trust obtained from the network. In the case of `fromLedger`, we do always need to verify down to that ledger as we do not have a trusted hash for its corresponding checkpoint. 

That said, I can widen the bounds if we want to be extra sure that the hashes match.